src/client/article.ts

```diff
@@ -214,6 +214,45 @@ function hasSeenCards(): void {
     })
 }
 
+function safeParse(item: {} | string): any {
+    try {
+        const str = typeof item === 'string' ? item : JSON.stringify(item);
+        return JSON.parse(str);
+    } catch(e) {
+        return {};
+    }
+}
+
+function iframeHeightChanges() {
+    const figures = Array.from(document.querySelectorAll('figure.interactive'));
+    figures.forEach(figure => {
+        const serverRenderedIframe = figure.querySelector('iframe');
+        if (!serverRenderedIframe) return;
+        const iframe = document.createElement('iframe');
+        iframe.style.width = '100%';
+        iframe.style.border = 'none';
+        iframe.height = '500';
+        iframe.src = serverRenderedIframe.src;
+
+        window.addEventListener('message', function(event) {
+            if (event.source !== iframe.contentWindow) {
+                return;
+            }
+
+            const message = safeParse(event.data);
+
+            switch (message.type) {
+                case 'set-height':
+                    iframe.height = message.value;
+                    break;
+                default:
+            }
+        }, false);
+
+        figure.replaceChild(iframe, serverRenderedIframe);
```

Comment:
What made you opt to replace the server-rendered `iframe` instead of attaching listeners for messages from it?

Reply:
The parent window receives messages via `postMessage` in the iframe to `set-height` but I couldn't work out how to match that request to a particular iframe.

We have the `origin` available in every request to update the height, but we would have a problem if we have two interactives from the same origin. The only solution I could find was to replace them all client side - this is how the templates currently handle them.